File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public string solution(int[] food) {
        string answer = "";
        List<int> foodList = new List<int>();
        List<int> foodElementList = new List<int>();
        foodList = food.ToList();
        for(int i = 1; i < foodList.Count; i++) {
            int amount = foodList[i]/2;
            for(int j = 0; j < amount; j++) {
                foodElementList.Add(i);
            }
        }
        answer = String.Join("", foodElementList);
        answer += "0";
        foodElementList.Reverse();
        answer += String.Join("",foodElementList);
        
        return answer;
    }
}

File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    var answer = '';
    const left = [];
    
    for (let i = 1; i <= food.length; i ++) {
        left.push((i+"").repeat(food[i]/2))
    }
    let word = left.join("") + "0";
    const right = left.reverse()
    word += right.join("")        
    return word;
}